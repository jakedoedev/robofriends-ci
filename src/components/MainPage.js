import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import Header from '../components/Header';

import ErrorBoundry from '../components/ErrorBoundry';
import './MainPage.css';

class MainPage extends Component{

  // Reduxed
  // constructor(){
  //   super();
  //   this.state = {
  //     robots: [],
  //     // searchField: '' // Reduxed
  //   }
  // }

  // Reduxed
  // onSearchChange = (event) => {
  //   this.setState({ searchField: event.target.value});
  // }

  componentDidMount(){
    this.props.onRequestRobots();

    // Reduxed
    // fetch('https://jsonplaceholder.typicode.com/users')
    // .then(response => response.json())
    // .then(users =>  this.setState({ robots: users}));
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;

    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
  }

  render(){
    const { onSearchChange, isPending } = this.props;

    if(isPending){
      return <h1 className="tc"> Loading </h1>;
    }

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          <ErrorBoundry>
            <CardList robots={this.filterRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
