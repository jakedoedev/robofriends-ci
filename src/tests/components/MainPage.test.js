import React from 'react';
import {shallow, mount, render} from 'enzyme';
import MainPage from '../../components/MainPage';
import { wrap } from 'module';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage { ...mockProps} />);
})

it('Renders MainPage component', () => {
  // expect(shallow(<MainPage />).length).toEqual(1);
  expect(wrap).toMatchSnapshot();
})

it('Displays pending state', () => {
  const mockPropsFilter = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'Chris',
      email: 'chris@gmail.com'
    }],
    searchField: '',
    isPending: true
  }

  const wrapperFiltered = shallow(<MainPage { ...mockPropsFilter} />);
  expect(wrapperFiltered.instance().props.isPending).toEqual(true);
  expect(wrapperFiltered.html()).toEqual('<h1 class="tc"> Loading </h1>');
});

it('Filters robots correctly', () => {
  expect(wrapper.instance().filterRobots()).toEqual([]);

  const mockPropsFilter = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'Chris',
      email: 'chris@gmail.com'
    }],
    searchField: 'c',
    isPending: false
  }

  const wrapperFiltered = shallow(<MainPage { ...mockPropsFilter} />);
  expect(wrapperFiltered.instance().filterRobots()).toEqual([{
    id: 3,
    name: 'Chris',
    email: 'chris@gmail.com'
  }]);
});

it('Returns empty if not found', () => {
  const mockPropsFilter = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'Chris',
      email: 'chris@gmail.com'
    }],
    searchField: 'z',
    isPending: false
  }
  const filteredRobots = [];
  const wrapperFiltered = shallow(<MainPage { ...mockPropsFilter} />);
  expect(wrapperFiltered.instance().filterRobots()).toEqual(filteredRobots);
});
