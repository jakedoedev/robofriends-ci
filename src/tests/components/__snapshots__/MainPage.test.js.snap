// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MainPage component 1`] = `[Function]`;
